src/vg/civcraft/mc/citadel/listener/BlockListener.java

```diff
@@ -79,13 +80,25 @@
 	public void onFortificationMode(BlockPlaceEvent event){
 		Player p = event.getPlayer();
 		Block b = event.getBlock();
+		Location loc = b.getLocation();
 		Inventory inv = p.getInventory();
+		Reinforcement rein = rm.getReinforcement(b.getLocation());
 		if (Material.AIR.equals(event.getBlockReplacedState().getType())) {
-			Reinforcement rein = rm.getReinforcement(b.getLocation());
 			if (rein != null && rein instanceof PlayerReinforcement){
+				//Would be nice to find a more performant way to detect the entity w/o checking every entity.
```

Comment:
We've run into this issue before, there is no better way than checking all entities.